### PR TITLE
IGAPP-532: MapLibreWeb

### PR DESCRIPTION
**Pull request by f1sh1918**
Test Route:
[Testumgebung](http://0.0.0.0:9000/testumgebung/de/locations)

This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Review by Taggotty — APPROVED** [review]
Tested on several browsers and devices on browserstack.

File: web/src/components/MapView.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { defaultViewportConfig, mapStyleUrl } from 'api-client/src/mapView'
+
+const MapView: React.FunctionComponent = () => {
+  const [viewport, setViewport] = useState({
+    ...defaultViewportConfig
+  })
+
+  return (
+    <div style={{ display: 'flex', justifyContent: 'center' }}>
+      <ReactMapGL {...viewport} onViewportChange={nextViewport => setViewport(nextViewport)} mapStyle={mapStyleUrl} />
```
Comment (klinzo) [inline web/src/components/MapView.tsx:14]:
```suggestion
      <ReactMapGL {...viewport} onViewportChange={setViewport} mapStyle={mapStyleUrl} />
```

File: web/src/components/MapView.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React, { useState } from 'react'
+import ReactMapGL from 'react-map-gl'
+import 'maplibre-gl/dist/maplibre-gl.css'
+
+import { defaultViewportConfig, mapStyleUrl } from 'api-client/src/mapView'
+
+const MapView: React.FunctionComponent = () => {
+  const [viewport, setViewport] = useState({
+    ...defaultViewportConfig
+  })
```
Comment (klinzo) [inline web/src/components/MapView.tsx:10]:
```suggestion
  const [viewport, setViewport] = useState(defaultViewportConfig)
```

File: api-client/src/mapView/index.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+interface MapViewViewport {
```
Comment (klinzo) [inline api-client/src/mapView/index.ts:1]: I think we do not use interfaces anywhere, so my personal preference would be to use types here as well. Maybe we should add an linting rule for this.

**Review by klinzo — APPROVED** [review]

**Comment by klinzo** [comment]
Tested on firefox and chrome.

File: api-client/src/mapView/index.ts (hunk 3 above)
Comment (f1sh1918) [reply api-client/src/mapView/index.ts:1]: we talked about personal preferences :D

Comment (klinzo) [reply api-client/src/mapView/index.ts:1]: Thats why I explicitly said personal preference :D I just think we should use either types or interfaces and not both.

**Pull request merged**